THEMING.md

```diff
@@ -0,0 +1,142 @@
+<!--docs:
+title: "Theming"
+layout: detail
+section: components
+excerpt: "Material Theming refers to the customization of your Material Design app to better reflect your product’s brand."
+-->
+
+# Theming
+
+Material Theming refers to the customization of your Material Design app to better reflect your product’s brand.
+
+## Overview
+
+### Terminology
+
+Our approach to theming relies on the relationships between the following concepts:
+
+1. Components
+2. Subsystem Schemes
+3. The Container Scheme
+4. Theming Extensions
+
+Components are expected to provide public APIs for a variety of parameters. An example of a component is [MDCButton](https://github.com/material-components/material-components-ios/tree/develop/components/Buttons).
+
+Subsystem schemes represent a set of opinionated properties that are intended to be mapped to component parameters. There is a scheme for each Material Theming subsystem. For example, there is a scheme for color, shape, and typography subsystems.
+
+The Container scheme represents a single configurable entity that is applicable to all themeable components. A container scheme consists of all of the subsystem schemes.
+
+Theming extensions are component extensions that, when invoked with a default container scheme, will configure a component to match the design system's defaults. When provided with subsystem schemes via a container scheme, the extension will map the subsystem scheme's values to the component’s parameters.
+
+### Sensible defaults, yet highly configurable
+
+By default, we try and give our components sensible values for all of their customizable properties, things such as `backgroundColor` or `titleFont`. You can use these defaults as a baseline, but we encourage you to theming your components to match your brand style.
+
+### Schemes
+
+<ul class="icon-list">
+<li class="icon-list-item icon-list-item--link"><a href="Container/">Container scheme</a></li>
```

Comment:
Do these links work? I would expect these to be anchor tags.  I don't see the leading hash (`#`) and I don't see anchors (`<a name="Container" />`) being defined in the document.

Reply:
I don't know where we want to put this so until that is resolved the links won't work.

Reply:
Is this required for this PR? If it's the only blocker, can we remove the list and submit without it?  It looks like the docs site generates a table of contents if the README.md has the marker at the top of the file:

```html
<!-- toc -->
```

https://material.io/develop/ios/components/bottomappbar/

Reply:
Should we wait to make sure we place it at the correct location? I can fix the links, my only blocker is where do we put this? I am waiting for your (@romoore) question to be answered "Please let us know where website docs should live."

Reply:
Ping @jverkoey

Reply:
These links will work if we move this file to `components/schemes`.

Reply:
How will these work?
1. There's no document called "Container"
2. The href doesn't start with an anchor marker (`#`)
3. Regardless of (1) or (2), the relevant section is named "Container scheme" and not "Container".

Reply:
Ah, sorry. I see now—they point to subdirectories.  Yep, that makes much more sense if it's moved to `components/schemes/`